controller/query_blackbox_test.go

```diff
@@ -0,0 +1,347 @@
+			cq.Data.Attributes.Fields = `{"invalid: json"}`
+			svc, ctrl := rest.SecuredControllerWithIdentity(fxt.Identities[0])
+			// when
+			test.CreateQueryBadRequest(t, svc.Context, svc, ctrl, fxt.Spaces[0].ID, cq)
+		})
+		t.Run("conflict same title, spaceID, creator", func(t *testing.T) {
+			fxt := tf.NewTestFixture(t, rest.DB,
+				tf.CreateWorkItemEnvironment(),
+				tf.Queries(1, tf.SetQueryTitles("q1")))
+			cq := getQueryCreatePayload(fxt.Queries[0].Title, nil)
+			svc, ctrl := rest.SecuredControllerWithIdentity(fxt.Identities[0])
+			// when
+			test.CreateQueryConflict(t, svc.Context, svc, ctrl, fxt.Spaces[0].ID, cq)
+		})
+		t.Run("unknown space ID", func(t *testing.T) {
+			// given
+			fxt := tf.NewTestFixture(t, rest.DB, tf.CreateWorkItemEnvironment())
+			cq := getQueryCreatePayload("new query", nil)
+			svc, ctrl := rest.SecuredControllerWithIdentity(fxt.Identities[0])
+			// when
+			test.CreateQueryNotFound(t, svc.Context, svc, ctrl, uuid.NewV4(), cq)
+		})
+	})
+}
+
+func (rest *TestQueryREST) TestList() {
+	resetFn := rest.DisableGormCallbacks()
+	defer resetFn()
+
+	rest.T().Run("success", func(t *testing.T) {
+		t.Run("ok", func(t *testing.T) {
+			fxt := tf.NewTestFixture(t, rest.DB,
+				tf.CreateWorkItemEnvironment(),
+				tf.Queries(3, tf.SetQueryTitles("q1", "q2", "q3")))
+			fxt2 := tf.NewTestFixture(t, rest.DB,
+				tf.CreateWorkItemEnvironment(),
+				tf.Queries(3, tf.SetQueryTitles("q4", "q5", "q6")))
+			svc, ctrl := rest.SecuredControllerWithIdentity(fxt.Identities[0])
+			// when
+			_, qList := test.ListQueryOK(t, svc.Context, svc, ctrl, fxt.Spaces[0].ID, nil, nil)
+			// then
+			require.NotNil(t, qList)
+			mustHave := map[string]struct{}{
+				"q1": {},
+				"q2": {},
+				"q3": {},
+			}
+			for _, q := range qList.Data {
+				delete(mustHave, q.Attributes.Title)
+			}
+			assert.Empty(t, mustHave)
+			assert.Equal(t, 3, qList.Meta.TotalCount)
+			// list by different user
+			// when
+			svc, ctrl = rest.SecuredControllerWithIdentity(fxt2.Identities[0])
+			_, qList = test.ListQueryOK(t, svc.Context, svc, ctrl, fxt2.Spaces[0].ID, nil, nil)
+			// then
+			require.NotNil(t, qList)
+			mustHave = map[string]struct{}{
+				"q4": {},
+				"q5": {},
+				"q6": {},
+			}
+			for _, q := range qList.Data {
+				delete(mustHave, q.Attributes.Title)
+			}
+			assert.Empty(t, mustHave)
+			assert.Equal(t, 3, qList.Meta.TotalCount)
+		})
+	})
+
+	rest.T().Run("fail", func(t *testing.T) {
+		t.Run("unauthorized", func(t *testing.T) {
+			// given
+			fxt := tf.NewTestFixture(t, rest.DB,
+				tf.CreateWorkItemEnvironment(),
+				tf.Queries(2))
+			svc, ctrl := rest.UnSecuredController()
+			// when
+			test.ListQueryUnauthorized(t, svc.Context, svc, ctrl, fxt.Spaces[0].ID, nil, nil)
+		})
+		t.Run("unknown space ID", func(t *testing.T) {
+			// given
+			fxt := tf.NewTestFixture(t, rest.DB, tf.CreateWorkItemEnvironment())
+			svc, ctrl := rest.SecuredControllerWithIdentity(fxt.Identities[0])
```

Comment:
Since you're only using the identity I suggest you only create an Identity in the `fxt` setup above.

Reply:
sure! done in https://github.com/fabric8-services/fabric8-wit/pull/1824/commits/b7ae234e2887a5facf8b50ea655aa7dfa39996b2